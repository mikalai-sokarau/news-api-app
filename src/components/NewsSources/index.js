import React from "react";
import { Link } from "react-router-dom";
import { NEWS_SOURCES } from "../../common/constants";

const DEFAULT_NEWS = NEWS_SOURCES[0];

const NewsSources = ({ source } = DEFAULT_NEWS.shortName) => (
  <div className="collection">
    { NEWS_SOURCES.map(item => createLink(item, source)) }
  </div>
);

function createLink(item, source) {
  const active = source === item.shortName ? "active" : "";
  const linkClassName = `collection-item ${active}`;

  return (
    <Link
      to={{ pathname: `/${item.shortName}` }}
      key={item.name}
      className={linkClassName}
    >
      {item.name}
    </Link>
  );
}

export default NewsSources;
