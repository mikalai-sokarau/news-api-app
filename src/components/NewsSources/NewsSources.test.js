import React from 'react';
import { shallow } from 'enzyme';
import NewsSources from './index';

describe('<NewsSources />', () => {
  test('className should be `collection`', () => {
    expect(shallow(<NewsSources />)).toMatchSnapshot();
  });
});
