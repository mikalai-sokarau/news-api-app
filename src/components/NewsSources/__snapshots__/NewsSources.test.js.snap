// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsSources /> className should be \`collection\` 1`] = `
<div
  className="collection"
>
  <NewsSourcesLink
    item={
      Object {
        "name": "CNN",
        "query": "sources=cnn",
        "shortName": "cnn",
        "type": "top-headlines",
      }
    }
    key="CNN"
    source="cnn"
  />
  <NewsSourcesLink
    item={
      Object {
        "name": "The New York Times",
        "query": "sources=the-new-york-times",
        "shortName": "thenewyourktimes",
        "type": "top-headlines",
      }
    }
    key="The New York Times"
    source="cnn"
  />
  <NewsSourcesLink
    item={
      Object {
        "name": "MTV News",
        "query": "sources=mtv-news",
        "shortName": "mtvnews",
        "type": "top-headlines",
      }
    }
    key="MTV News"
    source="cnn"
  />
  <NewsSourcesLink
    item={
      Object {
        "name": "ESPN",
        "query": "sources=espn",
        "shortName": "espn",
        "type": "top-headlines",
      }
    }
    key="ESPN"
    source="cnn"
  />
</div>
`;
